airflow/www/app.py

```diff
@@ -34,6 +36,9 @@
 
 
 def create_app(config=None, testing=False):
+    # cleanup connections
+    signal.signal(signal.SIGTERM, sigterm_handler)
```

Comment:
Have you considered using https://docs.python.org/2/library/atexit.html here?  It seems to be more tailored for what you aim to achieve.